mobile/src/components/Report.js

```diff
@@ -34,69 +42,159 @@ export default class Report extends Component {
 
-      apiClient.getText(url)
-        .then((response) => {
-          this.setState({ reportHtml: response });
-        });
-    };
+    apiClient.get(url)
+      .then((response) => {
+        this.setState({ totalPages: Math.ceil(response[0].totalPages / this.state.pageSize) || 1 });
+      });
+  }
+
+  fetchReport = (pageNumber) => {
+    this.setState({ reportHtml: '' });
+    const offset = (pageNumber - 1) * this.state.pageSize;
+    const url = `/api/reports/templates/${this.state.reportId}/html?pageSize=20&offset=${offset}`;
+
+    apiClient.getText(url)
+      .then((response) => {
+        this.setState({ reportHtml: response });
+      });
+  };
+
+  fetchNextPage() {
+    const { currentPage } = this.state;
+    if (currentPage + 1 <= this.state.totalPages) {
+      this.fetchReport(currentPage + 1);
+      this.setState({ currentPage: currentPage + 1 });
+    }
+  }
 
-    fetchPdf() {
-      const url = `/api/reports/templates/${this.state.reportId}/pdf`;
-      apiClient.downloadReport(url, this.state.reportName, 'pdf');
+  fetchPrevPage() {
+    const { currentPage } = this.state;
+    if (currentPage - 1 >= 1) {
+      this.fetchReport(currentPage - 1);
+      this.setState({ currentPage: currentPage - 1 });
     }
+  }
 
-    fetchXls() {
-      const url = `/api/reports/templates/${this.state.reportId}/xls`;
-      apiClient.downloadReport(url, this.state.reportName, 'xls');
+  fetchFirstPage() {
+    const { currentPage } = this.state;
+    if (currentPage !== 1) {
+      this.fetchReport(1);
+      this.setState({ currentPage: 1 });
     }
+  }
 
-    render() {
-      return (
-        <View style={getContainerStyle()}>
-          <Text style={[styles.formHeader, lightThemeText]}>{this.state.reportName}</Text>
-
-          <View style={styles.buttonContainer}>
-            <Button
-              onPress={() => this.fetchPdf()}
-              iconName="download"
-              iconColor="#FFF"
-              buttonColor="#337ab7"
-            >
-                        Download PDF
-            </Button>
-            <Button
-              onPress={() => this.fetchXls()}
-              iconName="download"
-              iconColor="#FFF"
-              buttonColor="#449C44"
-              marginLeft={10}
-            >
-                        Download XLS
-            </Button>
-          </View>
-          <View style={reportStyles.reportHtmlStyle}>
-            <WebView
-              source={{ html: this.state.reportHtml }}
-              style={{ flex: 1 }}
-            />
-          </View>
-        </View>
-      );
+  fetchLastPage() {
+    const { currentPage } = this.state;
+    if (currentPage !== this.state.totalPages) {
+      this.fetchReport(this.state.totalPages);
+      this.setState({ currentPage: this.state.totalPages });
     }
+  }
+
+  fetchPdf() {
+    const url = `/api/reports/templates/${this.state.reportId}/pdf`;
+    apiClient.downloadReport(url, this.state.reportName, 'pdf');
```

Comment:
you will need to set the page size when downloading report pdf or xls, you can use max int as page size (2147483647)